index.src.html

```diff
@@ -769,103 +782,123 @@ <h3 id=interface-reporting-observer>Interface {{ReportingObserver}}</h3>
   like CSP reports will not be.  Maybe there's something about the settings
   object?
-  
+
   A <dfn id=dom-report interface>Report</dfn> is the application exposed
   representation of a <a>report</a>. <dfn attribute for="Report">type</dfn>
   returns [=report/type=], <dfn attribute for="Report">url</dfn> returns
-  [=report/url=] and <dfn attribute for="Report">body</dfn> returns
+  [=report/url=], and <dfn attribute for="Report">body</dfn> returns
   [=report/body=].
-    
+
   Each {{ReportingObserver}} object has these associated concepts:
-    - A <dfn for=ReportingObserver>callback</dfn> set on creation.
-    - An {{ReportingObserverOptions}} dictionary called
+    - A <dfn for=ReportingObserver>callback</dfn> function set on creation.
+    - A {{ReportingObserverOptions}} dictionary called
       <dfn for=ReportingObserver>options</dfn>.
     - A list of {{Report}} objects called the <dfn for=ReportingObserver>report
-      list</dfn> that is initially empty.
+      list</dfn>, which is initially empty.
 
   The <dfn constructor for=ReportingObserver><code>
   ReportingObserver(|callback|)</code></dfn> constructor, when invoked, must
   create a new {{ReportingObserver}} object with <a>callback</a> set to
   |callback|, and then return it.
-  
+
   The <dfn method for=ReportingObserver><code>observe(|options|)</code></dfn>
   method, when invoked, must run these steps:
-  
+
   1. Set the <a>context object</a>'s <a>options</a> to |options|.
-  
-  2. If it is not already present, add the <a>context object</a> to the 
-     list of <a>registered reporting observers</a> for the <a>environment
-     settings object</a> associated with the <a>context object</a>.
-  
+
+  2. Let |settings| be the <a>environment settings object</a> associated with
+     the <a>context object</a>
+
+  3. If it is not already registered, add the <a>context object</a> to
+     the <a>registered reporting observer list</a> of its
+     associated <a>environment settings object</a>.
+
+  4. If the <a>context object's</a> {{futureOnly}} <a>option</a> is true, stop.
+
+  5. For each <a>report</a> |report| in the <a>report buffer</a> associated with
+     |settings|, execute [[#add-report]] with |report| and the <a>context object</a>.
+
+  6. <a>Queue a task</a> to [[#invoke-observers]] with a list containing the <a>context object</a>
+
   The <dfn method for=ReportingObserver><code>disconnect()</code></dfn> method,
   when invoked, must run these steps:
-  
-  1. If present, remove the <a>context object</a> from the list of
-     <a>registered reporting observers</a> for the <a>environment settings
-     object</a> associated with the <a>context object</a>.
- 
+
+  1. If the <a>context object</a> is not <a>registered</a>, stop.
+
+  2. Remove the <a>context object</a> from the <a>registered reporting observer
+     list</a> of its associated <a>environment settings object</a>.
+
+  <h3 id="add-report" algorithm>
+    Add |report| to reporting |observer|
```

Comment:
It seems odd to use the term "reporting observer" here, which is a type, but then turn just the second word into a parameter name. Probably just call this algorithm "Add |report| to |observer|", and mention below that |observer| is a ReportingObserver

Reply:
Done.